### PR TITLE
add option in flexiprep to downsample input with a fraction

### PR DESCRIPTION
**Pull request by sndrtj**

### CODE REVIEW

File: flexiprep/src/main/scala/nl/lumc/sasc/biopet/pipelines/flexiprep/QcCommand.scala [hunk 1]
```diff
@@ -43,10 +43,20 @@ class QcCommand(val parent: Configurable, val fastqc: Fastqc, val read: String)
...
 
+  var downSampleFraction: Option[Float] =
+    config("downsample_fraction", namespace = "flexiprep", default = None)
+
   override def defaultCoreMemory = 2.0
   override def defaultThreads = 3
 
   val seqtk = new SeqtkSeq(parent)
+  val seqtkSample: Option[SeqtkSample] = downSampleFraction match {
+    case Some(f) if 0.0 < f && f < 1.0 =>
```
Comment (ffinfo) [inline flexiprep/src/main/scala/nl/lumc/sasc/biopet/pipelines/flexiprep/QcCommand.scala:55]: Maybe also add something like this after this case?
``` scala
case Some(f) => 
  Logging.addError("downsample_fraction should be more then 0.0 and lower then 1.0")
```

File: flexiprep/src/main/scala/nl/lumc/sasc/biopet/pipelines/flexiprep/QcCommand.scala [hunk 2]
```diff
@@ -99,7 +109,15 @@ class QcCommand(val parent: Configurable, val fastqc: Fastqc, val read: String)
   }
 
   override def beforeCmd(): Unit = {
-    seqtk.input = input
+    seqtkSample match {
+      case Some(subsample) =>
+        subsample.input = input
+        subsample.output = new File(output.getParentFile, input.getName + ".subsample.fq")
+        subsample.isIntermediate = true
```
Comment (ffinfo) [inline flexiprep/src/main/scala/nl/lumc/sasc/biopet/pipelines/flexiprep/QcCommand.scala]: This is executed inside a pipe so this is redundant

**Review by ffinfo — CHANGES_REQUESTED** [review]

**Comment by ffinfo** [comment]
An generic note, could you also add this to functional tests in flexiprep?

File: flexiprep/src/main/scala/nl/lumc/sasc/biopet/pipelines/flexiprep/QcCommand.scala (hunk 1 above)
Comment (sndrtj) [reply flexiprep/src/main/scala/nl/lumc/sasc/biopet/pipelines/flexiprep/QcCommand.scala:55]: :+1:

**Review by ffinfo — APPROVED** [review]

**Pull request merged**